substrate/frame/revive/fixtures/build.rs

```diff
@@ -116,50 +116,41 @@ mod build {
 		fs::write(output_dir.join("Cargo.toml"), cargo_toml).map_err(Into::into)
 	}
 
-	fn invoke_build(current_dir: &Path) -> Result<()> {
+	fn invoke_build(target: &Path, current_dir: &Path) -> Result<()> {
 		let encoded_rustflags = [
 			"-Dwarnings",
-			"-Ctarget-feature=+lui-addi-fusion",
-			"-Crelocation-model=pie",
-			"-Clink-arg=--emit-relocs",
-			"-Clink-arg=--unique",
 		]
 		.join("\x1f");
 
-		let toolchain = std::env::var(OVERRIDE_RUSTUP_TOOLCHAIN_ENV_VAR)
-			.ok()
-			.unwrap_or(String::from("rve-nightly"));
-
-		let build_res = Command::new(env::var("CARGO")?)
+		let mut build_command = Command::new(env::var("CARGO")?);
+		build_command
 			.current_dir(current_dir)
 			.env_clear()
 			.env("PATH", env::var("PATH").unwrap_or_default())
 			.env("CARGO_ENCODED_RUSTFLAGS", encoded_rustflags)
-			.env("RUSTUP_TOOLCHAIN", &toolchain)
 			.env("RUSTC_BOOTSTRAP", "1")
 			.env("RUSTUP_HOME", env::var("RUSTUP_HOME").unwrap_or_default())
 			.args([
 				"build",
 				"--release",
-				"--target=riscv32ema-unknown-none-elf",
 				"-Zbuild-std=core",
 				"-Zbuild-std-features=panic_immediate_abort",
 			])
-			.output()
-			.expect("failed to execute process");
+			.arg("--target")
```

Comment:
I just mention this for completeness since I did the `build.rs` research before I saw any of this:

- `build.rs` can print `cargo::rustc-link-arg` entries for passing extra linker arguments but this does not help with the compilation.
- The only possible (i.e. **untested**) way I found to pass a target, would be to do something along the lines of `println!("cargo::rustc-env=RUST_TARGET_PATH=<path>")`, where `<path>` is a colon separated list of directories. See https://doc.rust-lang.org/beta/nightly-rustc/rustc_target/spec/index.html. Then the target list in `rust-toolchain.xml` could be used to define the target list.

Not a real review comment but thought to write this down somewhere as "a study".

I randomly found also this fairly recent blog entry from Google for comparison: https://security.googleblog.com/2024/09/deploying-rust-in-existing-firmware.html

Pretty much the same same so I would not take great lengths to improve...

Reply:
I don't really get your point. Passing the target on the command lien works just fine?

Reply:
@koute asked me to look into `build.rs` so I did that. I approved the PR but wrote down what I found what can be done through `build.rs`.

Reply:
I.e. not exact a review comment just wanted put that down somewhere. I think the change is fine.